sdk/java/pulumi/src/main/java/com/pulumi/Pulumi.java

```diff
@@ -34,9 +35,45 @@ static void run(Consumer<Context> stack) {
      * @param stack the stack to run in Pulumi runtime
      * @return a future exit code from Pulumi runtime after running the stack
      * @see #run(Consumer)
+     * @see #withOptions(StackOptions)
      */
     static CompletableFuture<Integer> runAsync(Consumer<Context> stack) {
-        var pulumi = PulumiInternal.fromEnvironment();
-        return pulumi.runAsync(stack);
+        return withOptions(StackOptions.Empty).runAsync(stack);
+    }
+
+    /**
+     * @param options the {@link StackOptions} to use
+     * @return a Pulumi program entrypoint with given {@link StackOptions}
+     * @see #run(Consumer)
+     * @see #runAsync(Consumer)
+     */
+    static Pulumi.API withOptions(StackOptions options) {
+        return PulumiInternal.fromEnvironment(options);
+    }
+
+    /**
+     * Pulumi entrypoint operations.
+     */
+    interface API {
```

Comment:
we cannot reuse `Pulumi` interface with the same methods because they are static, but this actually may come in handy to add Automation API on top of

Reply:
A bit weird but OK with me 👍

Reply:
Yes, I agree, unfortunately that was the cleanest way I could think of to overcome the limitations in the language and have a nice API for the user.